src/main/java/com/adobe/analytics/client/auth/ClientCredentialsOAuthenticator.java

```diff
@@ -0,0 +1,73 @@
+package com.adobe.analytics.client.auth;
+
+import com.adobe.analytics.client.ConnectionUtil;
+import com.adobe.analytics.client.JsonUtil;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import org.apache.commons.codec.binary.Base64;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+public class ClientCredentialsOAuthenticator extends OAuthenticator {
+    private static final String OAUTH_URL = "https://api.omniture.com/token";
+
+    private final String clientId;
+
+    private final String clientSecret;
+
+    public ClientCredentialsOAuthenticator(String clientId, String clientSecret) {
+        this.clientId = clientId;
+        this.clientSecret = clientSecret;
+    }
+
+    @Override
+    public void getToken() throws JsonSyntaxException, IOException {
+        URL url = new URL(OAUTH_URL);
+        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+        setupPostRequest(conn);
+        final JsonObject response = JsonUtil.GSON.fromJson(ConnectionUtil.readResponse(conn),
+                JsonObject.class);
+        getTokenJSONResponse(response);
+    }
+
+    private void setupPostRequest(HttpURLConnection conn) throws IOException {
+        setDoOutput(conn);
+        setRequestMethod(conn);
+        setBasicAuth(conn);
+        writePostParameter(conn);
+    }
+
+    private void setDoOutput(HttpURLConnection conn) {
+        conn.setDoOutput(true);
+    }
+
+    private void setRequestMethod(HttpURLConnection conn) throws ProtocolException {
+        conn.setRequestMethod("POST");
+    }
+
+    private void setBasicAuth(HttpURLConnection conn) {
+        conn.setRequestProperty("Authorization", getAuthHeader());
+    }
+
+    private void writePostParameter(HttpURLConnection conn) throws IOException {
+        String urlParameters = "grant_type=client_credentials";
+        byte[] postData = urlParameters.getBytes(StandardCharsets.UTF_8);
+        int postDataLength = postData.length;
+        conn.setRequestProperty("Content-Length", Integer.toString(postDataLength));
+        try (DataOutputStream wr = new DataOutputStream(conn.getOutputStream())) {
```

Comment:
I'm not sure if the DataOutputStream is the right choice here. Maybe we can use the [`IOUtils.write`](https://github.com/Adobe-Marketing-Cloud/analytics-java-library/blob/master/src/main/java/com/adobe/analytics/client/AnalyticsClient.java#L57) method as in the AnalyticsClient?